delivery/media/watermarking/lib/media-delivery-watermarking.js

```diff
@@ -0,0 +1,2134 @@
+            var num = counter[0];
+            if (0 === num) throw new Error("hkdf: cannot expand more");
+            hmac.reset(), num > 1 && hmac.update(buffer), info && hmac.update(info), hmac.update(counter), 
+            hmac.finish(buffer), counter[0]++;
+        }
+        exports.HMAC = HMAC, exports.hash = hash, exports.default = hash, exports.hmac = hmac;
+        var hkdfSalt = new Uint8Array(exports.digestLength);
+        function hkdf(key, salt, info, length) {
+            void 0 === salt && (salt = hkdfSalt), void 0 === length && (length = 32);
+            for (var counter = new Uint8Array([ 1 ]), okm = hmac(salt, key), hmac_ = new HMAC(okm), buffer = new Uint8Array(hmac_.digestLength), bufpos = buffer.length, out = new Uint8Array(length), i = 0; i < length; i++) bufpos === buffer.length && (fillBuffer(buffer, hmac_, info, counter), 
+            bufpos = 0), out[i] = buffer[bufpos++];
+            return hmac_.clean(), buffer.fill(0), counter.fill(0), out;
+        }
+        function pbkdf2(password, salt, iterations, dkLen) {
+            for (var prf = new HMAC(password), len = prf.digestLength, ctr = new Uint8Array(4), t = new Uint8Array(len), u = new Uint8Array(len), dk = new Uint8Array(dkLen), i = 0; i * len < dkLen; i++) {
+                var c = i + 1;
+                ctr[0] = c >>> 24 & 255, ctr[1] = c >>> 16 & 255, ctr[2] = c >>> 8 & 255, ctr[3] = c >>> 0 & 255, 
+                prf.reset(), prf.update(salt), prf.update(ctr), prf.finish(u);
+                for (var j = 0; j < len; j++) t[j] = u[j];
+                for (j = 2; j <= iterations; j++) {
+                    prf.reset(), prf.update(u).finish(u);
+                    for (var k = 0; k < len; k++) t[k] ^= u[k];
+                }
+                for (j = 0; j < len && i * len + j < dkLen; j++) dk[i * len + j] = t[j];
+            }
+            for (i = 0; i < len; i++) t[i] = u[i] = 0;
+            for (i = 0; i < 4; i++) ctr[i] = 0;
+            return prf.clean(), dk;
+        }
+        exports.hkdf = hkdf, exports.pbkdf2 = pbkdf2;
+    }(exports);
+    var sha256 = exports.default;
+    for (var k in exports) sha256[k] = exports[k];
+    module.exports = sha256;
+}();
+
+class Watermarking {
+    constructor(wmOptions) {
+        if (this.wmOptions = wmOptions || {
+            tokenType: TokenType.CWT,
+            validateWMClaims: !0
+        }, this.wmOptions.tokenType = void 0 === this.wmOptions.tokenType ? TokenType.CWT : this.wmOptions.tokenType, 
+        this.wmOptions.validateWMClaims = void 0 === this.wmOptions.validateWMClaims || this.wmOptions.validateWMClaims, 
+        ![ TokenType.CWT, TokenType.JWT ].includes(this.wmOptions.tokenType)) throw new Error("Invalid token type! Only cwt or jwt auth tokens are supported!");
+        this.wmOptions.tokenType === TokenType.CWT ? this.cwtValidator = new CWTValidator(wmOptions) : this.jwtValidator = new JWTValidator(wmOptions);
+    }
+    async validateShortToken(authToken, keys) {
```

Comment:
Are short tokens part of the Dash-IF spec? If not, should they be in the public module?

Reply:
As of now, short tokens are specific as per to one of our customer. Not sure if they will be added to the dash-if spec in the future.

Reply:
AFAIU the expectation is that multiple customers will use this module. Including code that is specific to one customer doesn't make sense. 

Can the short code approach be abstracted out somehow? 
* Maybe the customer could pass a transformation function in? 
* Maybe there could be a function the customer could call that would do everything up to line 1834 but return the values for the rest of the validation?

Alternatively, the we could keep a version of the module with the customer-specific changes in house.

Reply:
I think best would be to separate validateShortToken function in different file and use it as per customer requirement.?

Reply:
That seems reasonable to me.